main.js

```diff
@@ -69,137 +74,82 @@ let end_screen = {
+    }
+}
 
-        let word_key = getWordKey();
-        let answer;
-        let correct;
-        let pressed_key = null;
+let maybe_backward_mask = {
+    timeline : [backward_mask],
+    conditional_function : function () {
+        let tvar = jsPsych.timelineVariable('backward_mask');
+        return typeof tvar === "string" && tvar.length > 0;
+    }
+}
 
-        if (data.response !== null) {
-            pressed_key = data.response.toUpperCase();
-        }
+let maybe_visual_target = {
+    timeline : [visual_target],
+    conditional_function : function () {
+        let tvar = jsPsych.timelineVariable('visual_target');
+        return typeof tvar === "string" && tvar.length > 0;
+    }
+}
 
-        // Add "static" information to output
-        data.condition = jsPsych.timelineVariable('item_type');
-        data.word = jsPsych.timelineVariable('word');
-        data.word_file = jsPsych.timelineVariable('wordfn');
-        data.prime = jsPsych.timelineVariable('prime');
-        data.prime_mask = jsPsych.timelineVariable('pmask');
-        data.id = jsPsych.timelineVariable('id');
-        data.expected_answer = jsPsych.timelineVariable('expected_answer');
-        data.useful_data_flag = true;
-
-        answer = pressed_key === word_key ? 1 : 0;
-        correct = answer === data.expected_answer;
-
-        // Add dynamic info to output.
-        data.pressed_key = pressed_key;
-        data.answer = answer;
-        data.correct = correct;
-        data.integer_correct = data.correct ? 1 : 0;
-        data.pressed_key = pressed_key;
-        data.yes_key = yes_key;
-        data.no_key = no_key;
+let maybe_auditory_target = {
+    timeline : [auditory_target],
+    conditional_function : function () {
+        let tvar = jsPsych.timelineVariable('auditory_target');
+        return typeof tvar === "string" && tvar.length > 0;
     }
-};
+}
 
-let present_feedback = {
-    type: jsPsychHtmlKeyboardResponse,
-    stimulus: function() {
-        let feedback_text = '<span style="color:red;font-size:30px;">Incorrect</span>';
-        let last_resp_acc = jsPsych.data.getLastTrialData().values()[0].correct;
-        if (last_resp_acc === true) {
-            feedback_text = '<span style="color:green;font-size:30px;">Correct!</span>';
-        }
-        return feedback_text;
-    },
-    choices: jsPsych.NO_KEYS,
-    trial_duration: FEEDBACK_DURATION
-};
+let trial_timeline = [
+    present_fixation,
+    maybe_forward_mask,
+    maybe_visual_prime,
+    maybe_auditory_prime,
+    maybe_backward_mask,
+    maybe_visual_target,
+    maybe_auditory_target
+];
 
 // (timeline) procedures //////////////////////////////////////////////////////////
 
 let practice_procedure = {
-    timeline:[
-        present_fixation,
-        present_prime_mask,
-        present_prime,
-        present_word,
-        present_feedback
-    ],
+    timeline:[].concat(trial_timeline, [present_feedback]),
```

Comment:
Could be written as `[...trial_timeline, present_feedback]`

Reply:
Ah, cool, I learned something new:), I've never seen the `...` operator.